### ISSUE TITLE
[BUG] Apalache litters working directory

### ISSUE THREAD

**Issue by shonfeder** [informalsystems/apalache#977]
Currently, running Apalache in a directory means you'll end up leaving behind a bunch of files which are usually ignored. This is noisy, and I'd expect users used to unixy tools to find this quite invasive. I suggest: 

- By default, we create these files in a directory in the [temp directory](https://stackoverflow.com/a/28689710/1187277) (ensuring they get cleaned up on restarts)
- Provide a CLI flag with which users can override the location where these files are to be created, also configurable by environment variable, so users can configure this behavior to behave any other way they'd like

The motivation to open this issue is thanks to @rodrigo7491 reporting his surprise at all the files that were created in the working tree after running the integration tests.

**Comment by konnov** [comment]
That makes sense. We have the directory `x/<pattern>` for this purpose, but some passes simply ignore it. Perhaps, we should introduce a singleton that returns the default directory for the clutter.

**Comment by konnov** [comment]
Now it is done like that:

```scala
    val outdir = options.getOrError("io", "outdir").asInstanceOf[Path]
```

The problem is that only preprocessing passes have access to the options. A singleton could be used from any place.

**Comment by shonfeder** [comment]
Alternatively, if we thought it important that this data be persistent across reboots, we could put it in a configurable cache, likely in a `~/.apalache` directory, as is now quite common with tools?

**Comment by konnov** [comment]
> the x directory is part of the litter, from my POV. This should also go into /tmp I think.

That is true. It is becoming a problem I have generated 56G in `x` yesterday :-)

I think a good solution would be to dump in `x` only if `--debug` is enabled, or maybe introduce another flag, like `-x` (for tracing). We should still keep this tracing in files, because it makes debugging much easier. I agree that we should make it disabled by default though.

**Comment by Kukovec** [comment]
Closed by #1036.
